Sources/MapboxNavigation/NavigationMapViewDelegate.swift

```diff
@@ -35,10 +35,10 @@ public protocol NavigationMapViewDelegate: class, UnimplementedLogging {
         
      Resulting `LineString` will then be styled using `NavigationMapView.navigationMapView(_:routeLineLayerWithIdentifier:sourceIdentifier:)` provided style or a default congestion style if above delegate method was not implemented.
      - parameter navigationMapView: The NavigationMapView.
-     - parameter route: The route that the sender is asking about.
+     - parameter routes: The routes that the sender is asking about.
      - returns: A `LineString` object that defines the shape of the route, or `nil` in case of default behavior.
      */
-    func navigationMapView(_ navigationMapView: NavigationMapView, shapeFor route: Route) -> LineString?
+    func navigationMapView(_ navigationMapView: NavigationMapView, shapeFor routes: [Route]) -> LineString?
```

Comment:
As I mentioned earlier this change is not needed. It should be possible to modify each route's shape, not all at once.

Reply:
So it should only modify and override one route, right? And it should be default as the mainRoute to be changed?

Reply:
I think so. As I understand `legIndex` was added to control specific leg visibility mostly in active guidance (and there is only one route at a time). I'd run Navigation SDK `0.40` and call `NavigationMapView.show()` method with certain legIndex and check whether all selected legs on those routes are highlighted. 

Maybe @1ec5 can provide more details as well?

Reply:
I see. I created a simple test case as the following for the SDK 0.4:
```
import UIKit
import MapboxDirections
import MapboxCoreNavigation
import MapboxNavigation

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        let origin = Waypoint(coordinate: CLLocationCoordinate2D(latitude: 38.9131752, longitude: -77.0324047))
        let median1 = Waypoint(coordinate: CLLocationCoordinate2D(latitude: 38.912, longitude: -77.032))
        let median2 = Waypoint(coordinate: CLLocationCoordinate2D(latitude: 38.911, longitude: -77.031))
        let median3 = Waypoint(coordinate: CLLocationCoordinate2D(latitude: 38.9, longitude: -77.03))
        let destination = Waypoint(coordinate: CLLocationCoordinate2D(latitude: 38.89, longitude: -77.029))

        let options = NavigationRouteOptions(waypoints: [origin, median1, median2, destination])

        Directions.shared.calculate(options) { [weak self] (session, result) in
                    switch result {
                    case .failure(let error):
                        print(error.localizedDescription)
                    case .success(let response):
                        guard let route = response.routes?.first, let strongSelf = self else {
                            return
                        }
                        
                        // For demonstration purposes, simulate locations if the Simulate Navigation option is on.
                        // Since first route is retrieved from response `routeIndex` is set to 0.
                        let navigationService = MapboxNavigationService(route: route, routeOptions: options, simulating: .always)
                        let navigationOptions = NavigationOptions(navigationService: navigationService)
                        let navigationViewController = NavigationViewController(for: route, routeOptions: options, navigationOptions: navigationOptions)
                        navigationViewController.modalPresentationStyle = .fullScreen
                        navigationViewController.mapView?.routeCasingColor = #colorLiteral(red: 0.9254902005, green: 0.2352941185, blue: 0.1019607857, alpha: 1)
                        navigationViewController.mapView?.trafficUnknownColor  = #colorLiteral(red: 0.3384324702, green: 0.6636026477, blue: 0.9891712385, alpha: 0.2964898768)
                        navigationViewController.mapView?.show([route], legIndex: 1)
                        

                        strongSelf.present(navigationViewController, animated: true, completion: nil)
                    }
        }
    }
}

```
And the screenshot is as following:
![show_test](https://user-images.githubusercontent.com/48976398/109239618-af967f00-778a-11eb-98d3-5e57df5cb3b7.png)

So the purpose of the ` navigationViewMapView?.show()` in 2.0 should be overriding specific route leg color (that's the one exposed)  of the main route, right? And could act as above screen shoot.

Reply:
And for the `navigationMapView(_ navigationMapView: NavigationMapView, shapeFor route: Route) -> LineString?` it used to be generating a `MGLShape` including the coordinates of the route and corresponding properties, including whether is's main route/ alternative route, whether it's wanted route leg, and the congestion level . But now in 2.0, we separate the `LineString` and `FeatureCollection`, if only returns the main route `LineString`. Should we still keep the delegate of this function? Now the line Layer is mainly controlled by the `FeatureCollection`.